components/steering_wheel/include/IO/IO_Digital.h

```diff
@@ -0,0 +1,58 @@
+/**
+ * IO_Digital.h
+ * Digital IO Module Header
+ * @note To transfer this module to another system, the digitalInput_E and the
+ *      typedef'd IO_Digital_Flag_t must be modified to be congruent with the new system
+ */
+
+#pragma once
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+#include "Types.h"
+#include "stm32f103xb.h" /**< Needed for GPIO_TypeDef */
+
+// System Includes
+#include <assert.h>
+
+
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+typedef enum
+{
+    SW1 = 0U,
+    SW2,
+    SW4,
+    SW5,
+    BTN_TOP_LEFT,
+    BTN_MIDDLE,
+    BTN_TOP_RIGHT,
+    NUM_INPUTS
+} digitalInput_E;
+
+typedef uint8_t IO_Digital_Flag_t; /**< Current Implementation can handle upto 64 inputs */
+static_assert(NUM_INPUTS <= (sizeof(IO_Digital_Flag_t) * 8), 
+    "Size of DigitalInput_E must be less than or equal to the width of IO_Digital_Flag_t");
+
+typedef struct
+{
+    GPIO_TypeDef* gpiox;
```

Comment:
`port` instead of `gpiox` please

Reply:
I looked through the CAN layer and I am having a bit of a hard time understanding really what the abstraction would be for this. I understand the benefit of abstracting away from GPIO_TypeDef, however when I try to think of the actual applicationg of it I only see currently
typedef GPIO_TypeDef IO_T;
and having the HAL_GPIO_ReadPin being abstracted though a function GPIO_ReadPin which would just call HAL_GPIO_ReadPin. Is this the idea you had in mind to create that abstraction?